src/os/bluestore/NVMEDevice.cc

```diff
@@ -679,7 +692,15 @@ void io_complete(void *t, const struct spdk_nvme_cpl *completion)
       }
       delete task;
     } else {
-      task->return_code = 0;
+      struct Task* primary = task->primary;
```

Comment:
done

Reply:
because `primary` is not used afterwards and it's not used in the `else` branch. one has to parse the code to look for the references of `primary` if its lexical scope is not limited in the "if" clause.

Reply:
OK, let me change it.